control-plane/connect-inject/terminatinggatewayservice_controller_test.go

```diff
@@ -0,0 +1,2407 @@
+				require.NoError(t, err)
+			}
+
+			// create the terminating gateway service controller.
+			tas := &TerminatingGatewayServiceController{
+				Client:       fakeClient,
+				Log:          logrtest.TestLogger{T: t},
+				ConsulClient: consulClient,
+				Scheme:       s,
+				AclEnabled:   tt.aclEnabled,
+			}
+
+			err = tas.updateStatus(context.Background(), tt.terminatingGatewayService)
+			require.NoError(t, err)
+
+			terminatingGatewayService := &v1alpha1.TerminatingGatewayService{}
+			terminatingGatewayServiceName := types.NamespacedName{
+				Name:      "Terminating gateway service-created",
+				Namespace: "default",
+			}
+			err = fakeClient.Get(context.Background(), terminatingGatewayServiceName, terminatingGatewayService)
+			require.NoError(t, err)
+
+			require.Equal(t, tt.expStatus.ServiceInfoRef.ServiceName, terminatingGatewayService.ServiceInfoRef().ServiceName)
+			require.Equal(t, tt.expStatus.ServiceInfoRef.PolicyName, terminatingGatewayService.ServiceInfoRef().PolicyName)
+			require.Equal(t, tt.expStatus.Conditions[0].Message, terminatingGatewayService.Status.Conditions[0].Message)
+
+		})
+	}
+}
+
+func TestTerminatingGatewayServiceUpdateStatusError(t *testing.T) {
+	nodeName := "test-node"
+	cases := []struct {
+		name                      string
+		terminatingGatewayService *v1alpha1.TerminatingGatewayService
+		reconcileErr              error
+		expStatus                 v1alpha1.TerminatingGatewayServiceStatus
+		aclEnabled                bool
+	}{
+		{
+			name: "updates status when there's no existing status",
+			terminatingGatewayService: &v1alpha1.TerminatingGatewayService{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "Terminating gateway service-created",
+					Namespace: "default",
+				},
+				Spec: v1alpha1.TerminatingGatewayServiceSpec{
+					Service: &v1alpha1.CatalogService{
+						Node:        "legacy_node",
+						Address:     "10.20.10.22",
+						ServiceName: "TerminatingGatewayServiceController-created-service",
+						ServicePort: 9003,
+					},
+				},
+			},
+			reconcileErr: errors.New("this is an error"),
+			expStatus: v1alpha1.TerminatingGatewayServiceStatus{
+				Conditions: v1alpha1.Conditions{
+					{
+						Type:    v1alpha1.ConditionSynced,
+						Status:  corev1.ConditionFalse,
+						Reason:  "InternalError",
+						Message: "this is an error",
+					},
+				},
+			},
+			aclEnabled: true,
+		},
+		{
+			name: "updates status when there is an existing status",
+			terminatingGatewayService: &v1alpha1.TerminatingGatewayService{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "Terminating gateway service-created",
+					Namespace: "default",
+				},
+				Spec: v1alpha1.TerminatingGatewayServiceSpec{
+					Service: &v1alpha1.CatalogService{
+						Node:        "legacy_node",
+						Address:     "10.20.10.22",
+						ServiceName: "TerminatingGatewayServiceController-created-service",
+						ServicePort: 9003,
+					},
```

Comment:
This case should have an existing status

Reply:
Done